metadata/gc.go

```diff
@@ -111,6 +125,8 @@ type referenceLabelHandler struct {
 }
 
 func startGCContext(ctx context.Context, collectors map[gc.ResourceType]Collector) *gcContext {
+	ctx, span := tracing.StartSpan(ctx, tracing.Name(gcSpanPrefix, "startGCContext"))
```

Comment:
Do we need a new span here? Based on multiple feedback, creation of new spans should mark entry/exit into a component,  for internal methods we can use the existing span in context (using `SpanFromContext()`) and add attributes/events accordingly. We can make exceptions based on the usecase, say while tracing methods that are known to have high latencies or go routines etc. In this scenario, I don't think creating a new span adds much value. Applies to other places as well. 

Also should keep in mind that some trace services charge by the number of spans, so adding a lot of spans might not be feasible